plugin/collector/snap-collector-mock1/mock/mock_small_test.go

```diff
@@ -0,0 +1,220 @@
+
+	Convey("Testing CollectMetric", t, func() {
+
+		newPlg := new(Mock)
+		So(newPlg, ShouldNotBeNil)
+
+		Convey("with 'test' config variable'", func() {
+
+			node := cdata.NewNode()
+			node.AddItem("test", ctypes.ConfigValueBool{Value: true})
+			cfg := plugin.ConfigType{ConfigDataNode: node}
+
+			Convey("testing specific metrics", func() {
+				mTypes := []plugin.MetricType{
+					plugin.MetricType{Namespace_: ns0, Config_: cfg.ConfigDataNode},
+					plugin.MetricType{Namespace_: ns1, Config_: cfg.ConfigDataNode},
+					plugin.MetricType{Namespace_: ns2, Config_: cfg.ConfigDataNode},
+				}
+				mts, _ := newPlg.CollectMetrics(mTypes)
+
+				for _, mt := range mts {
+					_, ok := mt.Data_.(string)
+					So(ok, ShouldBeTrue)
+				}
+			})
+
+			Convey("testing specified metics", func() {
+				mTypes := []plugin.MetricType{
+					plugin.MetricType{Namespace_: ns3, Config_: cfg.ConfigDataNode},
+				}
+				mts, _ := newPlg.CollectMetrics(mTypes)
+
+				for _, mt := range mts {
+					_, ok := mt.Data_.(int)
+					So(ok, ShouldBeTrue)
+				}
+			})
+
+		})
+
+		Convey("without config variables", func() {
+
+			node := cdata.NewNode()
+			cfg := plugin.ConfigType{ConfigDataNode: node}
+
+			Convey("testing specific metrics", func() {
+				mTypes := []plugin.MetricType{
+					plugin.MetricType{Namespace_: ns0, Config_: cfg.ConfigDataNode},
+					plugin.MetricType{Namespace_: ns1, Config_: cfg.ConfigDataNode},
+					plugin.MetricType{Namespace_: ns2, Config_: cfg.ConfigDataNode},
+				}
+				mts, _ := newPlg.CollectMetrics(mTypes)
+
+				for _, mt := range mts {
+					_, ok := mt.Data_.(string)
+					So(ok, ShouldBeTrue)
+				}
+			})
+
+			Convey("testing specified metics", func() {
+				mTypes := []plugin.MetricType{
+					plugin.MetricType{Namespace_: ns3, Config_: cfg.ConfigDataNode},
+				}
+				mts, _ := newPlg.CollectMetrics(mTypes)
+
+				for _, mt := range mts {
+					_, ok := mt.Data_.(int)
+					So(ok, ShouldBeTrue)
+				}
+			})
+
+		})
+	})
+}
+
+func TestGetMetricTypes(t *testing.T) {
+	Convey("Tesing GetMetricTypes", t, func() {
+
+		newPlg := new(Mock)
+		So(newPlg, ShouldNotBeNil)
+
+		Convey("with missing on-load plugin config entry", func() {
+			node := cdata.NewNode()
+			node.AddItem("test-fail", ctypes.ConfigValueStr{Value: ""})
+
+			_, err := newPlg.GetMetricTypes(plugin.ConfigType{ConfigDataNode: node})
+
+			So(err, ShouldNotBeNil)
+		})
+
+		Convey("with 'test' config variable", func() {
+			node := cdata.NewNode()
+			node.AddItem("test", ctypes.ConfigValueStr{Value: ""})
+
+			mts, err := newPlg.GetMetricTypes(plugin.ConfigType{ConfigDataNode: node})
+
+			So(err, ShouldBeNil)
+			So(len(mts), ShouldEqual, 4)
+
+			Convey("checking namespaces", func() {
+				metricNames := []string{}
+				for _, m := range mts {
+					metricNames = append(metricNames, m.Namespace().String())
+				}
+
+				ns := core.NewNamespace("intel", "mock", "test")
```

Comment:
So this will lead, that most of our unit test will need to be ```medium```, because amount of mocking  gymnastics which we will be forced to do to achieve the goal of being tagged as ```small``` will be just to big to accept. I think we should rethink that....